pkg/cmd/extension/manager.go

```diff
@@ -172,7 +179,80 @@ func (m *Manager) InstallLocal(dir string) error {
 	return makeSymlink(dir, targetLink)
 }
 
-func (m *Manager) Install(cloneURL string, stdout, stderr io.Writer) error {
+type BinManifest struct {
+	Owner string
+	Name  string
+	Host  string
+	// TODO I may end up not using this; just thinking ahead to local installs
+	Path string
+}
+
+func (m *Manager) InstallBin(client *http.Client, repo ghrepo.Interface) error {
+	var r *release
+	r, err := fetchLatestRelease(client, repo)
+	if err != nil {
+		return err
+	}
+
+	suffix := m.platform()
+	var asset *releaseAsset
+	for _, a := range r.Assets {
+		if strings.HasSuffix(a.Name, suffix) {
+			asset = &a
+			break
+		}
+	}
+
+	if asset == nil {
+		return fmt.Errorf("%s unsupported for %s. Open an issue: `gh issue create -R%s/%s -t'Support %s'`",
+			repo.RepoName(),
+			suffix, repo.RepoOwner(), repo.RepoName(), suffix)
+	}
+
+	name := repo.RepoName()
+	targetDir := filepath.Join(m.installDir(), name)
+	// TODO clean this up if function errs?
+	err = os.MkdirAll(targetDir, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create installation directory: %w", err)
+	}
+
+	binPath := filepath.Join(targetDir, name)
+
+	err = downloadAsset(client, *asset, binPath)
+	if err != nil {
+		return fmt.Errorf("failed to download asset %s: %w", asset.Name, err)
+	}
+
+	manifest := BinManifest{
+		Name:  name,
+		Owner: repo.RepoOwner(),
+		Host:  repo.RepoHost(),
+		Path:  binPath,
+	}
+
+	bs, err := yaml.Marshal(manifest)
+	if err != nil {
+		return fmt.Errorf("failed to serialize manifest: %w", err)
+	}
+
+	manifestPath := filepath.Join(targetDir, "manifest.yml")
```

Comment:
This is interesting. When we discussed having a manifest file for binary extensions I envisioned we would have a single manifest file for all binary extensions, maybe even including git extensions, that would be the source of truth for what is installed. This way we could read one file rather than navigating the filesystem. Thoughts on the trade off of that vs having a manifest file for each installed extension?

Reply:
I'd rather do a manifest per extension. I don't immediately see a lot to gain from having a centralized manifest file for all extensions and it technically means that we would need to implement file locking since it becomes a database of sorts with race conditions (unlikely but it would be irresponsible to not take into account).